transform.ts

```diff
@@ -0,0 +1,45 @@
+import { Expr, Type } from "./ast";
+
+export function transformComprehension(comprehension: Expr<Type>): Expr<Type> {
+  if (comprehension.tag !== "comprehension") {
+    throw new Error("Expected a comprehension expression");
+  }
+
+  // TODO: Fully implement this function once we have some of the other parts of the compiler working
+  // Ensure iter: is a function call to range
+  if (comprehension.iter.tag !== "call" || comprehension.iter.name !== "range") {
```

Comment:
Good hack to get started. I wonder if you could give a type to comprehensions and let this be more general, and support other values next.

Reply:
Oh ok, so maybe instead of `comprehension.iter` being an `Expr<A>`, we can instead have it be something like `ComprehensionIterable<A>`, where this `ComprehensionIterable` type can be a list, etc.